cloudmarker/test/test_filestore.py

```diff
@@ -0,0 +1,69 @@
+"""Tests for filestore plugins."""
+
+
+import os
+import pathlib
+import shutil
+import unittest
+
+from cloudmarker.stores import filestore
+
+
+class FileStoreTest(unittest.TestCase):
+    """Tests for filestore plugins."""
+
+    @classmethod
+    def write_to_file(cls, filename):
+        # Initialize the FileStore plugin with the filename
+        store = filestore.FileStore(filename)
+
+        # dummy data to write
+        mp = {
+            "record_type": "dummy",
+            "data": {
+                "test_name": "tilde_filename"
+            }
+        }
+
+        store.write(mp)
+        store.done()
+
+    def test_filepath_tilde(self):
+        self.write_to_file("~/caramel_l33t/cloudmarker")
+        # if the parsing fails, it should create a "~" directory in the current
+        # path
+        f = pathlib.Path("~")
+
+        # If exists and is a directory, the parsing failed.
+        self.assertFalse(f.exists() and f.is_dir())
+
+    def test_filepath_tilde2(self):
+        user = os.getenv("USER")
+        self.write_to_file("~{}/caramel_tpb/cloudmarker".format(user))
+        # if the parsing fails, it should create a "~" directory in the current
+        # path
+        f = pathlib.Path("~{}".format(user))
+
+        # If exists and is a directory, the parsing failed.
+        self.assertFalse(f.exists() and f.is_dir())
+
+    def test_filepath_dir_traversal(self):
+        self.write_to_file("/tmp/cloudmarker/../abc")
+
+        f = pathlib.Path("/tmp/abc")
+        self.assertTrue(f.exists() and f.is_dir())
+
+    @classmethod
+    def tearDownClass(cls):
+        # remove all junk files created
+        user = os.getenv("USER")
+        p = [
+            "/tmp/abc",
+            "~/caramel_l33t",
+            os.path.join("~{}".format(user), "caramel_l33t")
```

Comment:
There should be one more directory here `caramel_tpb` as in the 2nd test method. And why caramel actually?

Reply:
Agreed, I had missed that folder.
`caramel` is a random string came to my mind at the time since I thought this test should not delete any actual folders. To make it more random, I added another suffix `tpb` which is the shorthand for [The Pirate Bay](https://en.wikipedia.org/wiki/The_Pirate_Bay) or more appropriately (the actual source) [TPB AFK (2013)](https://www.imdb.com/title/tt2608732/).